### PR TITLE
New delete method, edit method and exceptions

### PR DESCRIPTION
**Pull request by davidfreixes**
Related tasks: #6, #7, #16.

We have made the following modifications:

- editTask function has been added that modifies the name of a specific task. This task is identified by its ID. For this reason, a class has been added in Requests for this function (EditNameRequests.java)
- deleteTask function has been added that deletes a specific task. This task is also identified by its ID.
- We have modified the readme so that our EndPoints can be clearly seen.
- Added exception checking in Controller and Service.

Pending exceptions: getAll tasks, getTaskById and deleteTask.

### CODE REVIEW

**Review by IzarReina — APPROVED** [review]

**Comment by anniakuz** [comment]
The exceptions will be added in the next commit

File: src/main/java/todolist_grupo3/controllers/TaskController.java [hunk 1]
```diff
@@ -26,21 +31,39 @@ public class TaskController {
     @PostMapping("/task")
     @CrossOrigin("*")
     public ResponseEntity<?> createTask(@RequestBody CreateTaskRequest createTaskRequest) {
-        if (!createTaskRequest.isValid()) {
-            throw new RuntimeException("The name cannot be empty and cannot exceed 20 characters");
+        try {
+            return ResponseEntity.status(HttpStatus.CREATED).body(taskService.createTask(createTaskRequest.getName()));
+        } catch (HttpException e) {
```
Comment (SaraVictorFernandez) [inline src/main/java/todolist_grupo3/controllers/TaskController.java:36]: why should the service throw an HttpException?

File: src/main/java/todolist_grupo3/controllers/TaskController.java [hunk 2]
```diff
@@ -26,21 +31,39 @@ public class TaskController {
...
+    public ResponseEntity<?> getAllTasks() {
         return ResponseEntity.status(HttpStatus.OK).body(taskService.getAllTasks());
     }
 
     @GetMapping("/task/{id}")
     @CrossOrigin("*") 
-        public ResponseEntity<?> getTaskById(@PathVariable Integer id) {
-            return ResponseEntity.status(HttpStatus.OK).body(taskService.getTaskById(id));
+    public ResponseEntity<?> getTaskById(@PathVariable Integer id) {
+        Task task = taskService.getTaskById(id);
```
Comment (SaraVictorFernandez) [inline src/main/java/todolist_grupo3/controllers/TaskController.java:50]: what happens if the task does not exist?

File: src/main/java/todolist_grupo3/controllers/TaskController.java [hunk 3]
```diff
@@ -26,21 +31,39 @@ public class TaskController {
...
+            return ResponseEntity.status(HttpStatus.OK).body(task);
+    }
+
+    @DeleteMapping ("/task/{id}")
+    @CrossOrigin("*")
+    public ResponseEntity<?> deleteTask(@PathVariable Integer id) {
+        return ResponseEntity.status(HttpStatus.OK).body("Task deleted");
```
Comment (SaraVictorFernandez) [inline src/main/java/todolist_grupo3/controllers/TaskController.java:57]: who deletes the task?

File: src/main/java/todolist_grupo3/services/TaskServiceImpl.java [hunk 4]
```diff
@@ -18,6 +20,9 @@ public class TaskServiceImpl implements TaskService{
 
     @Override
     public Task createTask(String name) {
+        if (name == null || name.trim().isEmpty() || name.length() > 20) {
+            throw new HttpException(HttpStatus.BAD_REQUEST, "Error: name must be between 1 and 20 characters");
```
Comment (SaraVictorFernandez) [inline src/main/java/todolist_grupo3/services/TaskServiceImpl.java:24]: why is a service throwning http exceptions?

**Review by TAlonsoCueva — APPROVED** [review]

**Review by anniakuz — APPROVED** [review]

File: src/main/java/todolist_grupo3/services/TaskServiceImpl.java [hunk 5]
```diff
@@ -34,4 +39,21 @@ public Task getTaskById(Integer id){
...
     }
 
+    @Override
+    public void deleteTask (Integer id) {
+        taskRepository.deleteById(id);
+    }
+
+    @Override
+    public Task editTask(Integer id, String name) {
+        taskRepository.findById(id).orElseThrow(() -> new HttpException(HttpStatus.NOT_FOUND,
```
Comment (SaraVictorFernandez) [inline src/main/java/todolist_grupo3/services/TaskServiceImpl.java:49]: Why does the service throw an http exception?

**Review by SaraVictorFernandez — COMMENTED** [review]
Let's discuss a little bit about architecture

File: src/main/java/todolist_grupo3/controllers/TaskController.java (hunk 2 above)
Comment (davidfreixes) [reply src/main/java/todolist_grupo3/controllers/TaskController.java]: An exception has been added for this

File: src/main/java/todolist_grupo3/controllers/TaskController.java (hunk 1 above)
Comment (davidfreixes) [reply src/main/java/todolist_grupo3/controllers/TaskController.java:40]: has been changed to controller

**Review by SaraVictorFernandez — APPROVED** [review]
lgtm

**Pull request merged**